umh-core/pkg/config/config.go

```diff
@@ -18,14 +18,16 @@ import (
 	"reflect"
 
 	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/benthosserviceconfig"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/dataflowcomponentconfig"
 	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config/s6serviceconfig"
 
 	"github.com/tiendc/go-deepcopy"
 )
 
 type FullConfig struct {
-	Agent    AgentConfig    `yaml:"agent"`    // Agent config, requires restart to take effect
-	Internal InternalConfig `yaml:"internal"` // Internal config, not to be used by the user, only to be used for testing internal components
+	Agent    AgentConfig               `yaml:"agent"`    // Agent config, requires restart to take effect
```

Comment:
Nit: As i did on the redpanda pr, i recommend using omitempty, to ensure that the on disk config does not contain empty entries, which:
a) might confuse the user
b) might make gitops more difficult

Reply:
Will do